ProcessLib/ThermoHydroMechanics/CreateThermoHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,311 @@
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "THERMO_HYDRO_MECHANICS");
+    DBUG("Create ThermoHydroMechanicsProcess.");
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__HYDRO_MECHANICS__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    auto process_variables = findProcessVariables(
+        variables, pv_config,
+        {//! \ogs_file_param_special{prj__prcoesses__process__THERMO_HYDRO_MECHANICS__process_variables__temperature}
+         "temperature",
+         //! \ogs_file_param_special{prj__prcoesses__process__THERMO_HYDRO_MECHANICS__process_variables__pressure}
+         "pressure",
+         //! \ogs_file_param_special{prj__prcoesses__process__THERMO_HYDRO_MECHANICS__process_variables__displacement}
+         "displacement"});
+
+    DBUG("Associate displacement with process variable \'%s\'.",
+         process_variables[2].get().getName().c_str());
+
+    if (process_variables[2].get().getNumberOfComponents() != DisplacementDim)
+    {
+        OGS_FATAL(
+            "Number of components of the process variable '%s' is different "
+            "from the displacement dimension: got %d, expected %d",
+            process_variables[2].get().getName().c_str(),
+            process_variables[2].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    DBUG("Associate pressure with process variable \'%s\'.",
+         process_variables[1].get().getName().c_str());
+    if (process_variables[1].get().getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "Pressure process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            process_variables[1].get().getName().c_str(),
+            process_variables[1].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    DBUG("Associate temperature with process variable \'%s\'.",
+         process_variables[0].get().getName().c_str());
+    if (process_variables[0].get().getNumberOfComponents() != 1)
+    {
+        OGS_FATAL(
+            "Pressure process variable '%s' is not a scalar variable but has "
+            "%d components.",
+            process_variables[0].get().getName().c_str(),
+            process_variables[0].get().getNumberOfComponents(),
+            DisplacementDim);
+    }
+
+    // Constitutive relation.
+    // read type;
+    auto const constitutive_relation_config =
+        //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__constitutive_relation}
+        config.getConfigSubtree("constitutive_relation");
+
+    auto const type =
+        //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__constitutive_relation__type}
+        constitutive_relation_config.peekConfigParameter<std::string>("type");
+
+    std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
```

Comment:
This comment is not  for this PR:

It is better to have a class and its creator for mechanical properties to avoid redundant source code among mechanics process related classes.